numba/tests/test_listobject.py

```diff
@@ -37,6 +37,25 @@ def foo(n):
             self.assertEqual(foo(i), i)
 
 
+class TestAllocation(MemoryLeakMixin, TestCase):
+
+    def test_list_allocation(self):
+        @njit
+        def foo_kwarg(n):
+            l = listobject.new_list(int32, allocated=n)
+            return l._allocated()
+
+        for i in range(16):
```

Comment:
what about if the allocation is specified as negative?

Reply:
has been fixed and test added.